packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -272,6 +271,15 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
     for (let i = 0; i < renderedRows.length; i += 1) {
       const { id, model } = renderedRows[i];
 
+      let isSelected: boolean;
+      if (selectedRowsLookup[id] == null) {
+        isSelected = false;
+      } else if (typeof rootProps.isRowSelectable === 'function') {
+        isSelected = rootProps.isRowSelectable(apiRef.current.getRowParams(id));
```

Comment:
I'm not sure if this change is good. In the previous way, the selection lookup was already filtered all at once and memoized. Now `isRowSelectable` might be called several times for the same row. I'm OK to call it for every `currentPage.rows` and memoize the result.

Reply:
My thought behind this is that `renderedRows` will never be hundreds of rows long whereas `currentPage.rows` might have 100 000 items.
But I could memoize `renderedRows` and create a memoized selection lookup based on `renderedRows`.
That way we don't loop on the potentially very long `currentPage.rows` and we only re-run when the 1st and/or last  rendered rows changes.